frontend/src/components/ingress/Details.tsx

```diff
@@ -1,12 +1,157 @@
+import { Box, Link, Typography } from '@material-ui/core';
 import { useTranslation } from 'react-i18next';
 import { useParams } from 'react-router-dom';
 import Ingress, { IngressRule } from '../../lib/k8s/ingress';
+import { IngressBackend } from '../../lib/k8s/ingress';
 import { useSettings } from '../App/Settings/hook';
 import LabelListItem from '../common/LabelListItem';
 import { DetailsGrid } from '../common/Resource';
 import { SectionBox } from '../common/SectionBox';
 import SimpleTable from '../common/SimpleTable';
 
+/**
+ * This function is used to determine the protocol used in the formatURL for LinkedStringFormat
+ */
+function isHttpsUsed(item: Ingress, url: String) {
+  const hostList: string[] = item.jsonData.spec?.tls?.map(({ ...hosts }) => `${hosts.hosts}`);
+  const isHttps = hostList.includes(`${url}`);
+
+  return isHttps;
+}
+
+export interface LinkStringFormatProps {
+  url: string;
+  item: Ingress;
+  urlPath?: string;
+}
+
+/**
+ * This function is used to format the url to be used in the Link component
```

Comment:
Maybe get rid of this part: "This function is used to" which is a bit redundant for this component.